### PR TITLE
Perform cargo audit on nightly builds & do CI in github actions

### PR DESCRIPTION
**Pull request by jensim**
Hi again @seladb =)
I beleive we talked briefly about doing cargo audit in CI. Well now I've set it up for my little project over at https://github.com/jensim/bitbucket_server_cli and i think it works great so far. 
Wanna try it out?
Best regards
Jens

### CODE REVIEW

**Comment by seladb** [comment]
Thanks @jensim for letting me know!

If I understand this correctly `cargo audit` will only run vulnerable check but won't automatically fix it, is that correct?

Is there any way to get an alert when the audit fails? Otherwise we need to monitor the audit manually

**Comment by jensim** [comment]
Failed "builds" in github actions trigger notifications to the code owners, unless changed in the repo settings
If you want to fix it i believe `cargo audit fix` will do what it can for you. Cargo audit doesn't claim to be all knowing, but it's something.

**Comment by seladb** [comment]
Sure that makes sense.

If we're already doing this change, do you think we can move the entire CI to GitHub Actions? I don't see a point in maintaining 2 CI workflows (Travis and GitHub Actions). What do you think?

**Comment by jensim** [comment]
Yea that would be nice to give a try. @carllerche said the same thing about the builds for tokio-rs that currently run on Azure. `pickle-db`s conf for travis is not very big. I could give it a go. Separate PR maybe? Or do you prefer all-at-once lifting?

**Comment by seladb** [comment]
I think we can have both in the same PR. As you said it's not a lot of code. You can put them in separate commits so it'll be easier to track the changes

**Comment by jensim** [comment]
So i tried github actions out
https://github.com/jensim/pickledb-rs/actions/runs/103511375
I added a `continue-on-error: true` clause to `clippy`, since you have warnings there, and I assume that you were aware of those.
I opted for writing the config a bit airy, not to mix in to many `if`s here and there.

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    steps:
+      - uses: hecrj/setup-rust-action@v1
+        with:
+          rust-version: ${{ matrix.rust }}
+      - uses: actions/checkout@master
+      - name: Build
+        run: cargo build --examples --lib --bins
+      - name: Test
+        run: cargo test --verbose --examples --lib --bins
```
Comment (seladb) [inline .github/workflows/rust.yml:25]: why do we need to separate between `cargo build` and `cargo test`?

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        rust: [stable, nightly, beta]
+
+    steps:
+      - uses: hecrj/setup-rust-action@v1
+        with:
+          rust-version: ${{ matrix.rust }}
+      - uses: actions/checkout@master
```
Comment (seladb) [inline .github/workflows/rust.yml:21]: does it checkout only the `master` branch? We probably want the CI to work on any branch...

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        run: cargo test --verbose --examples --lib --bins
+  clippy:
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+      - uses: actions/checkout@master
+      - name: Install clippy
+        run: rustup component add clippy
+      - name: Run clippy
+        run: cargo clippy --all-targets -- --deny clippy::all
```
Comment (seladb) [inline .github/workflows/rust.yml:34]: clippy should run on rust stable branch only

File: .github/workflows/rust.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  fmt:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust: [stable, beta]
+    steps:
+      - uses: hecrj/setup-rust-action@v1
+        with:
+          rust-version: ${{ matrix.rust }}
+      - uses: actions/checkout@master
```
Comment (seladb) [inline .github/workflows/rust.yml:44]: does it checkout only the `master` branch? We probably want the CI to work on any branch...

File: .github/workflows/rust.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Build
+        run: cargo build --examples --lib --bins
+      - name: Test
+        run: cargo test --verbose --examples --lib --bins
+  clippy:
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+      - uses: actions/checkout@master
```
Comment (seladb) [inline .github/workflows/rust.yml:30]: does it checkout only the `master` branch? We probably want the CI to work on any branch...

File: .github/workflows/rust.yml (hunk 3 above)
Comment (jensim) [reply .github/workflows/rust.yml:34]: It does now

File: .github/workflows/rust.yml (hunk 1 above)
Comment (jensim) [reply .github/workflows/rust.yml:25]: Its to not get the super verbpse build output that comes from implicit build from running cargo test with verbose flag.

File: .github/workflows/rust.yml (hunk 5 above)
Comment (jensim) [reply .github/workflows/rust.yml:30]: Agree, really good point, I'll have a look at that

File: .github/workflows/rust.yml (hunk 4 above)
Comment (jensim) [reply .github/workflows/rust.yml]: That is a reference to version of the actions/checkout, i locked that down to v2 now, and have reconfigured the `on>push>[master]` to `on>push>[*]`

**Comment by jensim** [comment]
I've fixed it so that the 'Rust test' action is run on branches as well now

**Comment by jensim** [comment]
I removed the `continue-on-error: true` for the clippy step of the build, after I noticed that the travis-PR-build was failing there as well.
https://travis-ci.org/github/seladb/pickledb-rs/pull_requests

File: .github/workflows/rust.yml (hunk 3 above)
Comment (seladb) [reply .github/workflows/rust.yml]: got it 👍

File: .github/workflows/rust.yml (hunk 1 above)
Comment (seladb) [reply .github/workflows/rust.yml]: ok got it

File: .github/workflows/rust.yml (hunk 4 above)
Comment (seladb) [reply .github/workflows/rust.yml]: ok thanks

**Comment by seladb** [comment]
One more ask: is there a badge we can add to the README.md file to indicate CI status?

Maybe we can add 2 badges: one for the CI job (rust job) which runs after each commit, and one for the rust-audit which runs daily. what do you think?

**Comment by jensim** [comment]
> One more ask: is there a badge we can add to the README.md file to indicate CI status?
> 
> Maybe we can add 2 badges: one for the CI job (rust job) which runs after each commit, and one for the rust-audit which runs daily. what do you think?

Definitely! But we need the action ids, that are not available until we merge

----

Also, we should probably merge my second PR, that addresses the unused imports, that clippy complains about, first. Don't you think?

**Comment by seladb** [comment]
Thanks! would you like me to add these badges to README.md or you prefer to do it as part of the PR?

**Comment by jensim** [comment]
@seladb you can have a look at [this](https://help.github.com/en/github/administering-a-repository/enabling-required-status-checks), to see how to enable checks in PRs from the settings. If you'd like that ;-)

**Comment by seladb** [comment]
That's a good idea @jensim , thanks for your suggestion! I've created this rule for this repo

**Pull request merged**